l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,678 @@
+        }
+    }
+
+    /// @dev The receive function is called when ETH is sent directly to the contract.
+    receive() external payable {
+        // Do not expect to receive Ether directly, (weth bridging diasbled)
+        revert("ShB: direct ether transfer not allowed");
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ERA LEGACY FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Initiates a deposit by locking funds on the contract and sending the request
+    /// of processing an L2 transaction where tokens would be minted
+    /// only used for eth based chains
+    /// @dev If the token is bridged for the first time, the L2 token contract will be deployed. Note however, that the
+    /// newly-deployed token does not support any custom logic, i.e. rebase tokens' functionality is not supported.
+    /// @param _l2Receiver The account address that should receive funds on L2
+    /// @param _l1Token The L1 token address which is deposited
+    /// @param _mintValue The amount of baseTokens to be minted on L2. In this case Eth
+    /// @param _amount The total amount of tokens to be bridged
+    /// @param _l2TxGasLimit The L2 gas limit to be used in the corresponding L2 transaction
+    /// @param _l2TxGasPerPubdataByte The gasPerPubdataByteLimit to be used in the corresponding L2 transaction
+    /// @param _refundRecipient The address on L2 that will receive the refund for the transaction.
+    /// @dev If the L2 deposit finalization transaction fails, the `_refundRecipient` will receive the `_l2Value`.
+    /// Please note, the contract may change the refund recipient's address to eliminate sending funds to addresses
+    /// out of control.
+    /// - If `_refundRecipient` is a contract on L1, the refund will be sent to the aliased `_refundRecipient`.
+    /// - If `_refundRecipient` is set to `address(0)` and the sender has NO deployed bytecode on L1, the refund will
+    /// be sent to the `msg.sender` address.
+    /// - If `_refundRecipient` is set to `address(0)` and the sender has deployed bytecode on L1, the refund will be
+    /// sent to the aliased `msg.sender` address.
+    /// @dev The address aliasing of L1 contracts as refund recipient on L2 is necessary to guarantee that the funds
+    /// are controllable through the Mailbox, since the Mailbox applies address aliasing to the from address for the
+    /// L2 tx if the L1 msg.sender is a contract. Without address aliasing for L1 contracts as refund recipients they
+    /// would not be able to make proper L2 tx requests through the Mailbox to use or withdraw the funds from L2, and
+    /// the funds would be lost.
+    /// @return l2TxHash The L2 transaction hash of deposit finalization
+    function depositLegacyErc20Bridge(
+        address _msgSender,
+        address _l2Receiver,
+        address _l1Token,
+        uint256 _mintValue,
```

Comment:
Let's clean the interface. This function is only used in the `L1ERC20Bridge`. So not a public API.

`_mintValue` is enforced to be equal to the `msg.value`, but `L1ERC20Bridge` always sets `_mintValue` as `msg.value`. You can simply remove it then and use `msg.value` instead.